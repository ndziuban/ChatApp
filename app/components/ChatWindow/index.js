import React from 'react';
import PropTypes from 'prop-types';

import InputArea from '../InputArea';

class ChatWindow extends React.PureComponent {
  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onSubmit(msg) {
    this.props.onSubmit(this.props.id, msg)
  }

  render() {
    return (
      <div>
        {this.props.id} - Messages:
        <div>
          {
            this.props.messages.map(msg => <p>{msg.sender_id} says: {msg.text}</p>)
          }
        </div>
        <InputArea onSubmit={this.onSubmit}/>
      </div>
    );
  }
};

ChatWindow.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  messages: PropTypes.arrayOf(PropTypes.string),
  id: PropTypes.string.isRequired,
};

export default ChatWindow;
