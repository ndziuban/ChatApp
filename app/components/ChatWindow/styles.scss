.chat-window {
  width: 100%;
  align-self: flex-end;

  &__message-container {
    display: flex;
    flex-direction: column;
    padding: 5px 0;
  }
}
