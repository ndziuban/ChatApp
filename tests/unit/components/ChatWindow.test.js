import React from 'react';
import ChatWindow from '../../../app/components/ChatWindow';
import renderer from 'react-test-renderer';

it('ChatWindow (Not typing - Empty) renders correctly', () => {
  const tree = renderer
    .create(
      <ChatWindow
        sender="PERSON_1"
        receiver="PERSON_2"
        receiverIsTyping={false}
        messages={[]}
        onSubmit={jest.fn()}
        onSenderTyping={jest.fn()}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('ChatWindow (Typing - Empty) renders correctly', () => {
  const tree = renderer
    .create(
      <ChatWindow
        sender="PERSON_1"
        receiver="PERSON_2"
        receiverIsTyping
        messages={[]}
        onSubmit={jest.fn()}
        onSenderTyping={jest.fn()}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('ChatWindow (Not typing - Not Empty) renders correctly', () => {
  const tree = renderer
    .create(
      <ChatWindow
        sender="PERSON_1"
        receiver="PERSON_2"
        receiverIsTyping={false}
        messages={[{text: "Test", sender_id:"PERSON_1"}]}
        onSubmit={jest.fn()}
        onSenderTyping={jest.fn()}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('ChatWindow (Typing - Not Empty) renders correctly', () => {
  const tree = renderer
    .create(
      <ChatWindow
        sender="PERSON_1"
        receiver="PERSON_2"
        receiverIsTyping
        messages={[{text: "Test", sender_id:"PERSON_1"}]}
        onSubmit={jest.fn()}
        onSenderTyping={jest.fn()}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
