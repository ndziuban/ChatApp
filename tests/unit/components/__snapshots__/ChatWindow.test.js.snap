// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatWindow (Not typing - Empty) renders correctly 1`] = `
<div
  className="chat-window"
>
  <div
    className="chat-window__header"
  >
    <div
      className="header-content"
    >
      <span>
        PERSON_1
      </span>
      <span
        className="header-content__subtext"
      >
        Active
      </span>
    </div>
  </div>
  <div
    className="overflow-content"
  >
    <div
      className="chat-window__message-container"
    >
      <div />
    </div>
  </div>
  <div
    className="input__container"
  >
    <textarea
      className="input"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Type here to send a message..."
      rows="2"
      value=""
    />
  </div>
</div>
`;

exports[`ChatWindow (Not typing - Not Empty) renders correctly 1`] = `
<div
  className="chat-window"
>
  <div
    className="chat-window__header"
  >
    <div
      className="header-content"
    >
      <span>
        PERSON_1
      </span>
      <span
        className="header-content__subtext"
      >
        Active
      </span>
    </div>
  </div>
  <div
    className="overflow-content"
  >
    <div
      className="chat-window__message-container"
    >
      <div
        className="message  sender"
      >
        <span>
          Test
        </span>
      </div>
      <div />
    </div>
  </div>
  <div
    className="input__container"
  >
    <textarea
      className="input"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Type here to send a message..."
      rows="2"
      value=""
    />
  </div>
</div>
`;

exports[`ChatWindow (Typing - Empty) renders correctly 1`] = `
<div
  className="chat-window"
>
  <div
    className="chat-window__header"
  >
    <div
      className="header-content"
    >
      <span>
        PERSON_1
      </span>
      <span
        className="header-content__subtext"
      >
        Active
      </span>
    </div>
  </div>
  <div
    className="overflow-content"
  >
    <div
      className="chat-window__message-container"
    >
      <div
        className="message typing "
      >
        <span>
          PERSON_2 is typing...
        </span>
      </div>
      <div />
    </div>
  </div>
  <div
    className="input__container"
  >
    <textarea
      className="input"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Type here to send a message..."
      rows="2"
      value=""
    />
  </div>
</div>
`;

exports[`ChatWindow (Typing - Not Empty) renders correctly 1`] = `
<div
  className="chat-window"
>
  <div
    className="chat-window__header"
  >
    <div
      className="header-content"
    >
      <span>
        PERSON_1
      </span>
      <span
        className="header-content__subtext"
      >
        Active
      </span>
    </div>
  </div>
  <div
    className="overflow-content"
  >
    <div
      className="chat-window__message-container"
    >
      <div
        className="message  sender"
      >
        <span>
          Test
        </span>
      </div>
      <div
        className="message typing "
      >
        <span>
          PERSON_2 is typing...
        </span>
      </div>
      <div />
    </div>
  </div>
  <div
    className="input__container"
  >
    <textarea
      className="input"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Type here to send a message..."
      rows="2"
      value=""
    />
  </div>
</div>
`;
